src/commands/command.cr

```diff
@@ -73,8 +74,30 @@ module Shards
       Shards::Lock.write(packages, override_path, LOCK_FILENAME)
     end
 
-    def handle_resolver_errors(&)
+    private def log_available_tags(conflicts)
+      conflicts.join(separator: "\n") do |k, v|
+        req = v.requirement
+        tags = req.resolver.available_tags
+        req = req.requirement
+
+        if req.is_a?(Version) || (req.is_a?(VersionReq) && req.patterns.size == 1 && req.patterns[0] !~ /^(<|>|=)/)
+          req = "v" + req.to_s
+          found = Levenshtein.find(req, tags, 6)
+          "For #{k} the closest available tag to #{req} is: #{found}"
```

Comment:
**suggestion:** This branch is about version restrictions. We should not confuse that with tags.
It's an implementation detail that versions are represented as tags in the version control systems.

So the base set should be `available_releases` instead of `available_tags`. For a version restriction we don't care about all the tags, only the versions.

Also, I believe we need to explicitly handle the case that no versions are available.

Reply:
Do you mean simply replacing tags for release in messages and function names?

Reply:
The main change would be to use `resolver.available_releases` instead of `resolve.available_tags` as source of available items for this branch. And then adapt the wording as well, of course.

I can make a patch against your branch to demonstrate what I mean.

Reply:
But that won't work for what we want to do here. If you filter already the tags, then you don't get information to display the error

Reply:
I'll think of an alternative

Reply:
I use releases but also show tags

Reply:
Mmm... this is harder than thought:
 * Version is used in very specific cases, and I don't think a levenshtein diff will be useful
 * VersionReq has the "just this version" case which follows the patter of Version above. This is the only useful case.
 * Any other VersionReq is hard to comply to "close match". What is close match of ">0.3.0, <0.3.0"?
 * A failing branch fails at checkin, not at resolving, so won't be part of this case.
 * Tags never fail??

So I'm tempted to leave it as is for now.